core/test/matrix/batch_csr.cpp

```diff
@@ -152,6 +163,59 @@ TYPED_TEST(BatchCsr, CanBeEmpty)
 }
 
 
+TYPED_TEST(BatchCsr, CanBeDuplicatedFromOneCsrMatrix)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using value_type = typename TestFixture::value_type;
+    using index_type = typename TestFixture::index_type;
+    value_type values[] = {1.0, 2.0, 3.0, 4.0};
+    index_type col_idxs[] = {0, 1, 1, 0};
+    index_type row_ptrs[] = {0, 2, 3, 4};
+    value_type batch_values[] = {1.0, 2.0, 3.0, 4.0, 1.0, 2.0,
+                                 3.0, 4.0, 1.0, 2.0, 3.0, 4.0};
+
+    auto csr_mat = gko::matrix::Csr<value_type, index_type>::create(
+        this->exec, gko::dim<2>{3, 2},
+        gko::Array<value_type>::view(this->exec, 4, values),
+        gko::Array<index_type>::view(this->exec, 4, col_idxs),
+        gko::Array<index_type>::view(this->exec, 4, row_ptrs));
+
+    auto mtx = gko::matrix::BatchCsr<value_type, index_type>::create(
+        this->exec, 3, csr_mat.get());
+
+    this->assert_equal_data_array(12, batch_values, mtx->get_values());
+    this->assert_equal_data_array(4, col_idxs, mtx->get_col_idxs());
+    this->assert_equal_data_array(4, row_ptrs, mtx->get_row_ptrs());
+}
+
+
+TYPED_TEST(BatchCsr, CanBeDuplicatedFromBatchMatrices)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using value_type = typename TestFixture::value_type;
+    using index_type = typename TestFixture::index_type;
+    value_type values[] = {1.0, 2.0, 3.0, 4.0, -1.0, 12.0, 13.0, 14.0};
+    index_type col_idxs[] = {0, 1, 1, 0};
+    index_type row_ptrs[] = {0, 2, 3, 4};
+    value_type bvalues[] = {1.0, 2.0, 3.0, 4.0, -1.0, 12.0, 13.0, 14.0,
+                            1.0, 2.0, 3.0, 4.0, -1.0, 12.0, 13.0, 14.0,
+                            1.0, 2.0, 3.0, 4.0, -1.0, 12.0, 13.0, 14.0};
+
+    auto batch_mtx = gko::matrix::BatchCsr<value_type, index_type>::create(
+        this->exec, 2, gko::dim<2>{3, 2},
+        gko::Array<value_type>::view(this->exec, 8, values),
+        gko::Array<index_type>::view(this->exec, 4, col_idxs),
+        gko::Array<index_type>::view(this->exec, 4, row_ptrs));
+
+    auto mtx = gko::matrix::BatchCsr<value_type, index_type>::create(
+        this->exec, 3, batch_mtx.get());
+
+    this->assert_equal_data_array(24, bvalues, mtx->get_values());
```

Comment:
You could also test the size of the created matrix to make sure the metadata is populated correctly.

Reply:
I have added this now